packages/core/src/provider/__tests__/base.test.ts

```diff
@@ -102,6 +103,20 @@ describe("Base Tests", () => {
     `);
   });
 
+  it("should throw error if connected account has different entry point address than the provider", async () => {
```

Comment:
should this maybe throw in this case IFF the provider was initialized explicitly with an entrypoint?

that is if the provider init had undefined entrypoint it would inherit from the connected account

Reply:
good catch. yes